### PR TITLE
Add support for folding block/multiline comments

### LINKED ISSUE #21349: introduce drop down hiders for long comments
### Check for existing issues

- [X] Completed

### Describe the feature

image

comments with """ should get dropdowns just like functions get dropdowns

### Environment

Zed: v0.163.2 (Zed)
OS: Linux X11 linuxmint 22
Memory: 31.2 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by tomlin7**
Closes #21349 #19957 

Release Notes:

- Added multiline comments folding
- Added JS language config for block comments (used for demonstration)

https://github.com/user-attachments/assets/8a2698af-6edc-4167-9d0b-db2c5dafc6f6

Let me know if more configs need to be added as well! Also please correct me for the wrong approaches I have taken (if any)

### CODE REVIEW

**Comment by tomlin7** [comment]
@cla-bot check

File: crates/languages/src/javascript/config.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ path_suffixes = ["js", "jsx", "mjs", "cjs"]
 # [/ ] is so we match "env node" or "/node" but not "ts-node"
 first_line_pattern = '^#!.*\b(?:[/ ]node|deno run.*--ext[= ]js)\b'
 line_comments = ["// "]
+block_comment = ["/*", "*/"]
```
Comment (RemcoSmitsDev) [inline crates/languages/src/javascript/config.toml:7]: Hey, great work! You might want to add this also to the `Typescript` and `Tsx` languages.

**Comment by tomlin7** [comment]
@Angelk90 if language config specifies block comment delimitters then it would work just fine. I have only added config for JS atm.

**Comment by RemcoSmitsDev** [comment]
Just putting this out here, this is already possible for PHP. Might it be that we just need to configure it for the JavaScript language?

https://github.com/user-attachments/assets/ec10e3f2-6892-4293-a8d0-1844f5951ee9

File: crates/languages/src/javascript/config.toml (hunk 1 above)
Comment (tomlin7) [reply crates/languages/src/javascript/config.toml:7]: Thanks! I have added config for typescript and TSX as well

**Comment by RemcoSmitsDev** [comment]
> @RemcoSmitsDev : does it work like this in php too? Screenshot 2024-12-09 alle 16 17 37

No, because @tomlin7 just added this new action. But you can fold the multiline comment by just clicking inside the gutter, which does not yet work for JavaScript multiline comments.

**Comment by mikayla-maki** [comment]
> this is already possible for PHP

I believe that's because this comment is already indented one level, as we create the fold regions based purely on indentation level.

**Comment by mikayla-maki** [comment]
We would love to have this feature, but I don't think this implementation is the one we're looking for. Rather, I think this behavior should be integrated into our existing fold system so that the standard `editor::fold` action as well as the fold indicators in the gutter all smoothly interoperate with multiline comments.

However, this is such a large change in approach that I'm going to close this PR. If you decide to implement this, please open a new PR!

**Comment by RustoMCSpit** [comment]
> We would love to have this feature, but I don't think this implementation is the one we're looking for. Rather, I think this behavior should be integrated into our existing fold system so that the standard `editor::fold` action as well as the fold indicators in the gutter all smoothly interoperate with multiline comments.
> 
> However, this is such a large change in approach that I'm going to close this PR. If you decide to implement this, please open a new PR!

where is this behaviour being discussed? theres another issue here relevant to folding https://github.com/zed-industries/zed/issues/22373

**Pull request closed without merging**